From ed1575af3c98d89a4537a744c4c8382dfa951f74 Mon Sep 17 00:00:00 2001
From: Thor Thayer <thor.thayer@linux.intel.com>
Date: Thu, 30 Jul 2020 15:51:47 -0500
Subject: [PATCH 104/151] HSD #18012284911: arm64: dts: Fixes for DM Device
 Tree

commit  ce78963f74832bef5e0fe0dcb0f4322a0633d3a7 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Add fixed clocks for peripherals.
Move Simics specific code to the Simics DT.
Remove size restriction.
Add the SDMMC and enable the watchdog.

Signed-off-by: Thor Thayer <thor.thayer@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 arch/arm64/boot/dts/intel/socfpga_dm.dtsi     | 20 ++++++++++++++++++-
 .../boot/dts/intel/socfpga_dm_simics.dts      | 15 +++++++++++++-
 2 files changed, 33 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/socfpga_dm.dtsi b/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
index d40bf4d1701b..c2fda1ff7678 100644
--- a/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
+++ b/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
@@ -110,6 +110,7 @@
 			snps,multicast-filter-bins = <256>;
 			iommus = <&smmu 1>;
 			altr,sysmgr-syscon = <&sysmgr 0x44 0>;
+			clock-frequency = <250000000>;
 			status = "disabled";
 		};
 
@@ -126,6 +127,7 @@
 			snps,multicast-filter-bins = <256>;
 			iommus = <&smmu 2>;
 			altr,sysmgr-syscon = <&sysmgr 0x48 8>;
+			clock-frequency = <250000000>;
 			status = "disabled";
 		};
 
@@ -142,6 +144,7 @@
 			snps,multicast-filter-bins = <256>;
 			iommus = <&smmu 3>;
 			altr,sysmgr-syscon = <&sysmgr 0x4c 16>;
+			clock-frequency = <250000000>;
 			status = "disabled";
 		};
 
@@ -151,6 +154,7 @@
 			compatible = "snps,dw-apb-gpio";
 			reg = <0xffc03200 0x100>;
 			resets = <&rst GPIO0_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 
 			porta: gpio-controller@0 {
@@ -171,6 +175,7 @@
 			compatible = "snps,dw-apb-gpio";
 			reg = <0xffc03300 0x100>;
 			resets = <&rst GPIO1_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 
 			portb: gpio-controller@0 {
@@ -192,6 +197,7 @@
 			reg = <0xffc02800 0x100>;
 			interrupts = <0 103 4>;
 			resets = <&rst I2C0_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -202,6 +208,7 @@
 			reg = <0xffc02900 0x100>;
 			interrupts = <0 104 4>;
 			resets = <&rst I2C1_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -212,6 +219,7 @@
 			reg = <0xffc02a00 0x100>;
 			interrupts = <0 105 4>;
 			resets = <&rst I2C2_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -222,6 +230,7 @@
 			reg = <0xffc02b00 0x100>;
 			interrupts = <0 106 4>;
 			resets = <&rst I2C3_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -232,6 +241,7 @@
 			reg = <0xffc02c00 0x100>;
 			interrupts = <0 107 4>;
 			resets = <&rst I2C4_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -245,6 +255,7 @@
 			resets = <&rst SDMMC_RESET>;
 			reset-names = "reset";
 			iommus = <&smmu 5>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -271,6 +282,7 @@
 			#dma-requests = <32>;
 			resets = <&rst DMA_RESET>, <&rst DMA_OCP_RESET>;
 			reset-names = "dma", "dma-ocp";
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -310,6 +322,7 @@
 			resets = <&rst SPIM0_RESET>;
 			reg-io-width = <4>;
 			num-cs = <4>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -322,6 +335,7 @@
 			resets = <&rst SPIM1_RESET>;
 			reg-io-width = <4>;
 			num-cs = <4>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 		sysmgr: sysmgr@ffd12000 {
@@ -383,7 +397,7 @@
 			reg-shift = <2>;
 			reg-io-width = <4>;
 			resets = <&rst UART0_RESET>;
-			clock-frequency = <76800>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -403,6 +417,7 @@
 			reg = <0xffd00200 0x100>;
 			interrupts = <0 117 4>;
 			resets = <&rst WATCHDOG0_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -411,6 +426,7 @@
 			reg = <0xffd00300 0x100>;
 			interrupts = <0 118 4>;
 			resets = <&rst WATCHDOG1_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -419,6 +435,7 @@
 			reg = <0xffd00400 0x100>;
 			interrupts = <0 125 4>;
 			resets = <&rst WATCHDOG2_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
@@ -427,6 +444,7 @@
 			reg = <0xffd00500 0x100>;
 			interrupts = <0 126 4>;
 			resets = <&rst WATCHDOG3_RESET>;
+			clock-frequency = <100000000>;
 			status = "disabled";
 		};
 
diff --git a/arch/arm64/boot/dts/intel/socfpga_dm_simics.dts b/arch/arm64/boot/dts/intel/socfpga_dm_simics.dts
index f3f0485b1854..66dccaacaa1a 100644
--- a/arch/arm64/boot/dts/intel/socfpga_dm_simics.dts
+++ b/arch/arm64/boot/dts/intel/socfpga_dm_simics.dts
@@ -12,7 +12,7 @@
 	};
 
 	chosen {
-		bootargs = "mem=64M earlycon";
+		bootargs = "earlycon";
 		stdout-path = "serial0:4800n8";
 	};
 
@@ -24,5 +24,18 @@
 };
 
 &uart0 {
+	clock-frequency = <76800>;
+	status = "okay";
+};
+
+&mmc {
+	clock-frequency = <50000000>;
+	cap-sd-highspeed;
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&watchdog0 {
 	status = "okay";
 };
-- 
2.26.1

